spec/jobs/schedule_authorization_request_email_job_spec.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ScheduleAuthorizationRequestEmailJob do
+  describe '#perform' do
+    subject do
+      described_class.perform_now(
+        authorization_request_id,
+        authorization_request_status,
+        template_name,
+        recipients
+      )
+    end
+
+    let(:authorization_request_id) { authorization_request.id }
+    let(:authorization_request_status) { 'draft' }
+
+    let(:authorization_request) { create(:authorization_request, status: authorization_request_status) }
+    let(:to_user) { create(:user, :with_full_name) }
+
+    let(:template_name) { 'email_template' }
+    let(:vars) { {} }
+
+    let(:recipients) do
+      {
+        to: [
+          {
+            email: to_user.email,
+            full_name: to_user.full_name
+          }
+        ]
+      }
+    end
+
+    before do
+      class APIEntreprise::AuthorizationRequestMailer
+        def email_template; end
+      end
+    end
```

Comment:
```suggestion
    before(:all) do
      class APIEntreprise::AuthorizationRequestMailer
        def email_template; end
      end
    end
```

Y'avait vraiment besoin de redéfinir une méthode ? Elles sont toutes iso en plus donc en tester une = toutes les tester.

Reply:
Tu veux dire qu'on pourrait utiliser une méthode qui existe déjà dans le mailer? Je préfère en redéfinir une ici histoire de montrer que le fonctionnement est indépendant des méthodes existantes

Reply:
C'est peu conventionnel de ré-ouvre une classe existante dans la codebase pour y ajouter une méthode, ça peut avoir des side effects.

Je vais pas me formaliser là dessus, juste je ne comprends pas pourquoi on a besoin de définir une méthode supplémentaire (ce qui m'aménerai à scout commit si je repasse sur le fichier pour simplifier)